tests/src/test/scala/system/basic/WskConductorTests.scala

```diff
@@ -0,0 +1,311 @@
+      action.create(
+        echo,
+        Some(TestUtils.getTestActionFilename("echo.js")),
+        annotations = Map("conductor" -> true.toJson))
+    }
+
+    // a normal result
+    val run = wsk.action.invoke(echo, Map("payload" -> testString.toJson))
+    withActivation(wsk.activation, run) { activation =>
+      activation.response.status shouldBe "success"
+      activation.response.result shouldBe Some(JsObject("payload" -> testString.toJson))
+      checkConductorLogsAndAnnotations(activation, 1) // echo
+    }
+
+    // an error result
+    val secondrun = wsk.action.invoke(echo, Map("error" -> testString.toJson))
+    withActivation(wsk.activation, secondrun) { activation =>
+      activation.response.status shouldBe "application error"
+      activation.response.result shouldBe Some(JsObject("error" -> testString.toJson))
+      checkConductorLogsAndAnnotations(activation, 1) // echo
+    }
+
+    // a wrapped result { params: result } is unwrapped by the controller
+    val thirdrun = wsk.action.invoke(echo, Map("params" -> JsObject("payload" -> testString.toJson)))
+    withActivation(wsk.activation, thirdrun) { activation =>
+      activation.response.status shouldBe "success"
+      activation.response.result shouldBe Some(JsObject("payload" -> testString.toJson))
+      checkConductorLogsAndAnnotations(activation, 1) // echo
+    }
+
+    // an invalid action name
+    val invalidrun =
+      wsk.action.invoke(echo, Map("payload" -> testString.toJson, "action" -> invalid.toJson))
+    withActivation(wsk.activation, invalidrun) { activation =>
+      activation.response.status shouldBe "application error"
+      activation.response.result.get.fields.get("error") shouldBe Some(
+        JsString(compositionComponentInvalid(JsString(invalid))))
+      checkConductorLogsAndAnnotations(activation, 1) // echo
+    }
+
+    // an undefined action
+    val undefinedrun = wsk.action.invoke(echo, Map("payload" -> testString.toJson, "action" -> missing.toJson))
+    withActivation(wsk.activation, undefinedrun) { activation =>
+      activation.response.status shouldBe "application error"
+      activation.response.result.get.fields.get("error") shouldBe Some(JsString(compositionComponentNotFound(missing)))
+      checkConductorLogsAndAnnotations(activation, 1) // echo
+    }
+  }
+
+  it should "invoke a conductor action with dynamic steps" in withAssetCleaner(wskprops) { (wp, assetHelper) =>
+    val conductor = "conductor" // conductor action
+    assetHelper.withCleaner(wsk.action, conductor) { (action, _) =>
+      action.create(
+        conductor,
+        Some(TestUtils.getTestActionFilename("conductor.js")),
+        annotations = Map("conductor" -> true.toJson))
+    }
+
+    val step = "step" // step action
+    assetHelper.withCleaner(wsk.action, step) { (action, _) =>
+      action.create(step, Some(TestUtils.getTestActionFilename("step.js")), memory = Some(128 MB))
```

Comment:
is the 128MB here important? On the open deployment, I think 256MB is a stem cell whereas this would not be... not sure if you're trying to force a particular condition.

Reply:
Yes I am trying to test the max-memory-across-the-composition computation. I could change to a different value, other than 256 MB. What would be a better choice?

Reply:
perhaps a comment for clarity - otherwise fine then.